CAN_Decoder_SourceCode_TestRel_21Dec21.py

```diff
@@ -0,0 +1,107 @@
+import tkinter as tk
+from tkinter import *
+from tkinter import filedialog
+from datetime import datetime
+
+
+# Function for opening the file
+def CAN_Decoder():
+    lineNum = 0
+    oldDataFile = filedialog.askopenfile(initialdir="/")
+    listedOldData = []
+    for dataLine in oldDataFile:
+        segmentedDataLine = dataLine.split(" ")
+        listedOldDataLine = []
+        for stringSegment in segmentedDataLine:
+            if stringSegment.strip():
+                listedOldDataLine.append(stringSegment)
+        listedOldData.append(listedOldDataLine)
+        lineNum = lineNum + 1
+
+    newData = []
+    lineCount = len(listedOldData)
+    for dataLineIndex in range(0, lineCount):
```

Comment:
Why don't you iterate over `listedOldData` directly ? 
`for dataLine in listedOldData`

Reply:
lineCount is re-used while parsing data for consecutive frames